docs/site/Loopback-component-authentication.md

```diff
@@ -0,0 +1,611 @@
+---
+lang: en
+title: 'Authentication'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Loopback-component-authentication.html
+---
+
+## Overview
+
+Security is of paramount importance when developing a web or mobile application
+and usually consists of two distinct pieces:
+
+- Authentication
+- Authorization
+
+`Authentication` is a process of verifying someone's identity before a protected
+endpoint is accessed.
+
+`Authorization` is a process of verifying the actions that an authenticated user
+is allowed to perform on a protected endpoint.
+
+{% include note.html content="
+For a description of an `Authorization` process, please see [Authorization](Loopback-component-authorization.md).
+" %}
+
+This document describes the details of the LoopBack 4 `Authentication`
+component.
+
+Here is a **high level** overview of the authentication component.
+
+![authentication_overview_highlevel](./imgs/authentication_overview_highlevel.png)
+
+It is made up of a decorator, a few providers, and an extension point to which
+authentication strategies must be registered.
+
+Here is a **detailed** overview of the authentication component.
+
+![authentication_overview_detailed](./imgs/authentication_overview_detailed.png)
+
+Basically, to secure your API endpoints, you need to:
+
+- create/register a custom authentication strategy with a unique **name**
+- insert the authentication action in a custom sequence
+- decorate the endpoints of a controller with the
+  `@authenticate(strategyName, options?)` authentication decorator
+
```

Comment:
Maybe change the order as follows:
1. decorate (app developer)
2. add authentication action (app developer/cli)
3. create custom authentication strategy (extension developer)
4. register a custom authentication strategy (app developer)